src/internal/client/client.go

```diff
@@ -897,6 +897,24 @@ func (c *APIClient) AddMetadata(ctx context.Context) context.Context {
 	return metadata.NewOutgoingContext(ctx, finalMD)
 }
 
+func SetAuthToken(ctx context.Context, token string) context.Context {
```

Comment:
The GRPC way of doing this is dialing with a per-rpc credentials option; there is code floating around somewhere (I think we used it for hub and the debug dump service when we couldn't depend on pachyderm/pachyderm for dependency reasons).